src/main/kotlin/no/nav/henvendelsesarkiv/Database.kt

```diff
@@ -0,0 +1,93 @@
+package no.nav.henvendelsesarkiv
+
+import no.nav.henvendelsesarkiv.model.Arkivpost
+import no.nav.henvendelsesarkiv.model.ArkivpostMapper
+import no.nav.henvendelsesarkiv.model.Vedlegg
+import no.nav.henvendelsesarkiv.model.VedleggMapper
+import org.springframework.jdbc.core.JdbcTemplate
+
+private const val ARKIVPOST_SQL = """
+            INSERT INTO arkivpost(arkivpostId, arkivertDato, mottattDato, utgaarDato, temagruppe, arkivpostType, dokumentType,
+                                  kryssreferanseId, kanal, aktoerId, fodselsnummer, navIdent, innhold, journalfoerendeEnhet,
+                                  status, kategorikode, signert, erOrganInternt, begrensetPartInnsyn, sensitiv
+            ) VALUES (
+                ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?
+            )
+            """
+
+private const val VEDLEGG_SQL = """
+            INSERT INTO vedlegg(
+                arkivpostId, filnavn, filtype, variantformat, tittel, brevkode, strukturert, dokument
+            ) VALUES (
+                ?, ?, ?, ?, ?, ?, ?, ?
+            )
+            """
+
+class Database constructor(private val jt: JdbcTemplate, private val useHsql: Boolean = false) {
+
+    fun hentHenvendelse(id: Long): Arkivpost? {
+        val arkivpostSql = "SELECT * FROM arkivpost WHERE arkivpostId = ?"
+        val vedleggSql = "SELECT * FROM vedlegg WHERE arkivpostId = ?"
+        val arkivpost = jt.queryForObject(arkivpostSql, ArkivpostMapper(), id)
+        if(arkivpost != null) {
+            jt.query(vedleggSql, VedleggMapper(), id).forEach { arkivpost.vedleggListe.add(it) }
+        }
+        return arkivpost
+    }
+
+    fun opprettHenvendelse(arkivpost: Arkivpost): Long {
+        arkivpost.arkivpostId = nextSequenceValue()
+        jt.update(ARKIVPOST_SQL) {
+            setLong(it, 1, arkivpost.arkivpostId)
```

Comment:
Hva er disse tallene du sender med?

Reply:
det er indeks på "?" i prepared statement. litt dårlig konvensjon, men man har ikke noe valg hvis man bruker sql i stedet for orm/dsl

Reply:
Okei!